src/Services/BookingService/BookingService.cs

```diff
@@ -70,38 +70,47 @@ public class BookingService : IBookingService
             _httpContextAccessor = httpContextAccessor;
         }
 
-        public async Task<BookingProcessEntity> Get(string baseUrl, Page currentPage, string guid)
+        public async Task<BookingProcessEntity> Get(string baseUrl, Page currentPage, string sessionGuid)
         {
             var bookingElement = currentPage.Elements
-                .Where(_ => _.Type == EElementType.Booking)
-                .FirstOrDefault();
+                .First(element => element.Type
+                    .Equals(EElementType.Booking));
 
-            var appointmentTimes = new List<AvailabilityDayResponse>();
+            List<AvailabilityDayResponse> appointmentTimes = new();
 
             var bookingInformationCacheKey = $"{bookingElement.Properties.QuestionId}{BookingConstants.APPOINTMENT_TYPE_SEARCH_RESULTS}";
-            var cachedAnswers = _distributedCache.GetString(guid);
 
-            if (cachedAnswers != null)
-            {
-                var convertedAnswers = JsonConvert.DeserializeObject<FormAnswers>(cachedAnswers);
+            var appointmentType = bookingElement.Properties.AppointmentTypes
+                .GetAppointmentTypeForEnvironment(_environment.EnvironmentName);
 
+            var cachedAnswers = _distributedCache.GetString(sessionGuid);
+            FormAnswers convertedAnswers = new();
+            if (cachedAnswers is not null)
+            {
+                convertedAnswers = JsonConvert.DeserializeObject<FormAnswers>(cachedAnswers);
                 if (convertedAnswers.FormData.ContainsKey(bookingInformationCacheKey))
                 {
-                    var cachedBookingInformation = JsonConvert.DeserializeObject<BookingInformation>(convertedAnswers.FormData[bookingInformationCacheKey].ToString());
-                    var cachedInfo = new List<object> { cachedBookingInformation };
-                    return new BookingProcessEntity { BookingInfo = cachedInfo };
+                    var cachedBookingInformation = JsonConvert
+                        .DeserializeObject<BookingInformation>(convertedAnswers.FormData[bookingInformationCacheKey].ToString());
+
+                    if (appointmentType.AppointmentId.Equals(Guid.Empty))
+                        _mappingService.MapAppointmentId(appointmentType, convertedAnswers);
+
```

Comment:
we do the same call below aswell, is there a way to see if the user has defined a appointmentId, say on the model HasBookingIdFromQuestion => string.isnullorempty(AppointmentId) and if thats true we call _mappingService once?

Reply:
Added bool to AppointmentType model, and now calls "map" once from within cached data, as this can only be used when the booking element is on the second page at least, anyway...